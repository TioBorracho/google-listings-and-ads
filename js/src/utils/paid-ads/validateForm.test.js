/**
 * Internal dependencies
 */
import validateForm from './validateForm';

/**
 * `validateForm` function returns an object, and if any checks are not passed,
 * set properties respectively with an error message to indicate it.
 */
describe( 'validateForm', () => {
	let values;

	beforeEach( () => {
		// Initial values
		values = { countryCodes: [], amount: 0 };
	} );

	it( 'When all checks are passed, should return an empty object', () => {
		const errors = validateForm( {
			countryCodes: [ 'US' ],
			amount: 1,
		} );

		expect( errors ).toStrictEqual( {} );
	} );

	it( 'should indicate multiple unpassed checks by setting properties in the returned object', () => {
		const errors = validateForm( values );

		expect( errors ).toHaveProperty( 'countryCodes' );
		expect( errors ).toHaveProperty( 'amount' );
	} );

	it( 'When the country codes array is empty, should not pass', () => {
		const errors = validateForm( values );

		expect( errors ).toHaveProperty( 'countryCodes' );
		expect( errors.countryCodes ).toMatchSnapshot();
	} );

	it( 'When the amount is not a number, should not pass', () => {
		let errors;

		values.amount = '';
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();

		values.amount = undefined;
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();

		values.amount = new Date();
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();

		values.amount = NaN;
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();
	} );

	it( 'When the amount is â‰¤ 0, should not pass', () => {
		let errors;

		values.amount = 0;
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();

		values.amount = -0.01;
		errors = validateForm( values );

		expect( errors ).toHaveProperty( 'amount' );
		expect( errors.amount ).toMatchSnapshot();
	} );
} );
