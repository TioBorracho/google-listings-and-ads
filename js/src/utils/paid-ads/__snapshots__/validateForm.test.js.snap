// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateForm When the amount is not a number, should not pass 1`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the amount is not a number, should not pass 2`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the amount is not a number, should not pass 3`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the amount is not a number, should not pass 4`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the amount is ≤ 0, should not pass 1`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the amount is ≤ 0, should not pass 2`] = `"Please make sure daily average cost is greater than 0."`;

exports[`validateForm When the country codes array is empty, should not pass 1`] = `"Please select at least one country for your ads campaign."`;
